File: IBM_Bank/src/main/java/com/IBM/IBM_Bank/Models/Conta.java
package com.IBM.IBM_Bank.Models;

import javax.persistence.*;
import java.util.List;

@Entity
public class Conta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoConta tipoConta;  // Tipo da conta (e.g., "CORRENTE", "POUPANCA")

    @Column(nullable = false)
    private Double saldo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private StatusConta status;

    @OneToOne(mappedBy = "conta", cascade = CascadeType.ALL)
    private Cliente cliente;

    @OneToMany(mappedBy = "conta", cascade = CascadeType.ALL)
    private List<Movimentacao> movimentacoes;

    @OneToOne(mappedBy = "conta", cascade = CascadeType.ALL)
    private ContaCredito contaCredito;

    // Getters e Setters

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public TipoConta getTipoConta() {
        return tipoConta;
    }

    public void setTipoConta(TipoConta tipoConta) {
        this.tipoConta = tipoConta;
    }

    public Double getSaldo() {
        return saldo;
    }

    public void setSaldo(Double saldo) {
        this.saldo = saldo;
    }

    public StatusConta getStatus() {
        return status;
    }

    public void setStatus(StatusConta status) {
        this.status = status;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public List<Movimentacao> getMovimentacoes() {
        return movimentacoes;
    }

    public void setMovimentacoes(List<Movimentacao> movimentacoes) {
        this.movimentacoes = movimentacoes;
    }

    public ContaCredito getContaCredito() {
        return contaCredito;
    }

    public void setContaCredito(ContaCredito contaCredito) {
        this.contaCredito = contaCredito;
    }
}

File: IBM-bank-frontend/src/shared/models/conta.ts
export enum TipoConta {
    CORRENTE = 'CORRENTE',
    POUPANCA = 'POUPANCA'
}

export enum StatusConta {
    ATIVA = 'ATIVA',
    INATIVA = 'INATIVA'
}

export interface Conta {
    id: number;
    numero: number;
    tipoConta: TipoConta;
    saldo: number;
    status: StatusConta;
    clienteId: number;
}

export enum eColunaXPropConta {
    // 'Nome da coluna' = 'nome da propriedade'
    'ID' = 'id',
    'Número' = 'numero',
    'Tipo da Conta' = 'tipoConta',
    'Saldo' = 'saldo',
    'Status' = 'status',
    'ID do Cliente' = 'clienteId',
}